src/api/projectsAPI.ts

```diff
@@ -1,41 +1,143 @@
+  const projectInfo = await universalGet<ProjectGetResponse>(`/api/projects/${projectId}`);
+  return {
+    id: projectInfo.project_id,
+    path: projectInfo.project_path,
+    name: projectInfo.project_name,
+  };
 }
 
-export async function postRemoteProject(projectName: string) {
-  const response = await fetch(`/api/projects/?project_name=${projectName}`, {
-    method: 'POST',
-    // We need to update the API definition in order to use BODY
-    // https://stackoverflow.com/questions/59929028/python-fastapi-error-422-with-post-request-when-sending-json-data
-    // body: JSON.stringify({ project_name: projectName }),
-    headers: {
-      Accept: 'application/json',
-    },
-  });
-  const payload = (await response.json()) as ProjectsResponse;
+export async function createRemoteProject(projectName: string, projectPath?: string): Promise<ProjectInfo> {
+  const payload = await universalPost<ProjectPostResponse>(
+    `/api/projects/?project_name=${projectName}&project_path=${encodeURIComponent(projectPath ?? '')}`,
+    {},
+  );
   const projectId = Object.keys(payload)[0];
   const { project_path, project_name } = payload[projectId];
+
   return {
-    projectId,
-    projectPath: project_path,
-    projectName: project_name,
+    id: projectId,
+    path: project_path,
+    name: project_name,
   };
 }
+
+// ########################################################################################
+// PROJECT FILE STRUCTURE
+// ########################################################################################
+/** NOTE: This is a simplified version of the function */
+export async function readProjectFiles(projectId: string): Promise<TauriFileEntry[]> {
+  const projectInfo = await universalGet<ProjectGetResponse>(`/api/projects/${projectId}`);
+  const relativePaths = projectInfo.filepaths
+    .map((path) => path.replace(projectInfo.project_path + '/', '')) // Remove project path from API output
+    .map((path) => path.replace(/^\//, '')); // Remove the leading / (we don't want them in the output)
+
+  const rootFiles = relativePaths
+    .filter((path) => !path.startsWith('uploads'))
+    .map((path) => ({
+      path,
+      name: path.replace('/', ''),
+      children: undefined,
+    }));
+
+  const uploadsFiles = relativePaths
+    .filter((path) => path.startsWith('uploads/'))
+    .map((path) => ({
+      path,
+      name: path.replace('uploads/', ''),
+      children: undefined,
+    }));
+
+  return [
+    ...rootFiles,
+    {
+      path: 'uploads',
+      name: 'uploads',
+      children: uploadsFiles,
+    },
+  ];
+}
+
+// ########################################################################################
+// PROJECT FILE SYSTEM
+// ########################################################################################
+export const writeProjectTextFile = async (projectId: string, relativeFilePath: string, contents: string) => {
```

Comment:
Can simplify to:

```ts
export const writeProjectTextFile = async (projectId: string, relativeFilePath: string, contents: string) =>
  universalPutFile(`/api/fs/${projectId}/${relativeFilePath}`, relativeFilePath, contents);
```

and also for the next method, unless it's intentional that you want to swallow the return value of `universalPutFile`.

Reply:
Ok. Fixed this.